diff --git a/cloud/thinq2/device.ts b/cloud/thinq2/device.ts
--- a/cloud/thinq2/device.ts
+++ b/cloud/thinq2/device.ts
@@ -106,6 +106,10 @@ export class DeviceAcceptor extends TypedEmitter<DeviceAcceptorEvents> {
                         payload: JSON.stringify(generateDeployResponse(payload as ClipDeployMessage)),
                     },
                     null,
                 )
+
+                if (payload.cmd === 'deploy' && !client.deviceObj) {
+                    this.completeProvisioning(payload.did, payload, client)
+                }
             }
         }
     }
